app/uk/gov/hmrc/teamsandrepositories/persitence/TeamsAndReposPersister.scala

```diff
@@ -55,67 +52,76 @@ class TeamsAndReposPersister @Inject()(
     Future.traverse(teamNames)(mongoTeamsAndReposPersister.deleteTeam)
   }
 
-  def resetLastActiveDate(repoName: String)(implicit ec: ExecutionContext): Future[Option[Int]] =
+  def resetLastActiveDate(repoName: String)(implicit ec: ExecutionContext): Future[Option[Long]] =
     mongoTeamsAndReposPersister.resetLastActiveDate(repoName)
-
 }
 
 @Singleton
-class MongoTeamsAndRepositoriesPersister @Inject()(mongoConnector: MongoConnector, futureHelpers: FutureHelpers)
-    extends ReactiveRepository[TeamRepositories, BSONObjectID](
+class MongoTeamsAndRepositoriesPersister @Inject()(mongoComponent: MongoComponent, futureHelpers: FutureHelpers)(
+  implicit ec: ExecutionContext)
+    extends PlayMongoRepository(
+      mongoComponent = mongoComponent,
       collectionName = "teamsAndRepositories",
-      mongo          = mongoConnector.db,
-      domainFormat   = TeamRepositories.formats) {
+      domainFormat   = TeamRepositories.formats,
+      indexes        = Seq(IndexModel(Indexes.hashed("teamName"), IndexOptions().name("teamNameIdx")))
+    ) {
+  private val logger = Logger(this.getClass)
 
-  override def indexes: Seq[Index] =
-    Seq(Index(Seq("teamName" -> IndexType.Hashed), name = Some("teamNameIdx")))
+  def insert(teamAndRepos: TeamRepositories): Future[Boolean] =
+    collection.insertOne(teamAndRepos).toFuture().map(_.wasAcknowledged())
 
   def update(teamAndRepos: TeamRepositories)(implicit ec: ExecutionContext): Future[TeamRepositories] =
     futureHelpers
       .withTimerAndCounter("mongo.update") {
         collection
-          .update(ordered = false)
-          .one(
-            q      = Json.obj("teamName" -> Json.toJson(teamAndRepos.teamName)),
-            u      = teamAndRepos,
-            upsert = true)
+          .replaceOne(
+            filter      = equal("teamName", teamAndRepos.teamName),
+            replacement = teamAndRepos,
+            options     = ReplaceOptions().upsert(true)
+          )
+          .toFutureOption()
           .map(_ => teamAndRepos)
       }
       .recover {
         case lastError =>
           throw new RuntimeException(s"failed to persist $teamAndRepos", lastError)
       }
 
-  def getAllTeamAndRepos(archived: Option[Boolean])(implicit ec: ExecutionContext): Future[List[TeamRepositories]] = {
+  def getTeamsAndRepos(serviceNames: Seq[String])(implicit ec: ExecutionContext): Future[Seq[TeamRepositories]] =
+    collection
+      .find(
+        elemMatch(
+          "repositories",
+          or(
+            serviceNames
+              .map(serviceName => regex("name", "^" + serviceName + "$")): _*
```

Comment:
the reason for the regex matching was for the caseInsensitive "i" flag, which looks like it has been lost in this PR.
It would be preferable to use a caseInsensitive index, and then we don't need to do the regex stuff, and get performance.